db/migrate/20190226184255_check_guid_uniqueness.rb

```diff
@@ -0,0 +1,106 @@
+  end
+
+  class Host < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class LifecycleEvent < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class MiqAction < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class MiqAeWorkspace < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class MiqEventDefinition < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class MiqPolicy < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class MiqRegion < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class MiqServer < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class MiqSet < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class MiqWidget < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class MiqWorker < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class ScanItem < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class Service < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class ServiceTemplate < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class VmOrTemplate < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class Vm < ActiveRecord::Base
+    validates :guid, :uniqueness => true
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  def up
+    [AutomateWorkspace, ExtManagementSystem, Host, Job, LifecycleEvent, MiqAction, MiqAeWorkspace, MiqEventDefinition,
+     MiqPolicy, MiqRegion, MiqServer, MiqWidget, MiqWorker, ScanItem, Service, ServiceTemplate,
+     MiqSet, ScanItem, Vm].each do |thing|
+      say_with_time("Checking #{thing} for guid uniqueness") do
+        thing.where(:guid => thing.select(:guid).group(:guid).having("count(*) > 1").pluck(:guid))[0...-1].each { |obj| obj.guid = SecureRandom.uuid; obj.save! }
```

Comment:
I don't think you want the `pluck` in here, right?

Reply:
Also, the `[0...-1]` will leave out the last one.  I think we would actually want to leave the first record alone since it would be the original. (we may want to `order(:id)` to guarantee that)

Reply:
`.each { |obj| obj.guid = SecureRandom.uuid; obj.save! }` could be simplified to `.each { |obj| obj.update_attributes!(:guid => SecureRandom.uuid) }`